src/search/cost.cc

```diff
@@ -361,13 +609,32 @@ double cost::perf_cost(prog* synth, int len, bool set_win) {
   return total_cost;
 }
 
+//called by alpha
 double cost::total_prog_cost(prog * orig, int len1, prog * synth, int len2) {
   bool flag = (synth->_error_cost == -1);
-  double err_cost = error_cost(orig, len1, synth, len2);
-  double per_cost = perf_cost(synth, len2, true);
-  if (flag && logger.is_print_level(LOGGER_DEBUG)) {
-    cout << "cost: " << err_cost << " " << per_cost << " "
-         << (_w_e * err_cost) + (_w_p * per_cost) << endl;
+  double err_cost = 0.0;
+  double per_cost = 0.0;
+  double safe_cost = 0.0;
+
+  if (_functionality == 1){
+    //for repair
+    err_cost = error_cost_repair(orig, len1, synth, len2);
+    safe_cost = safety_cost_repair(orig, len1, synth, len2);
+
+    return (_w_e * err_cost) + (_w_s * safe_cost);
+
+  }else{
```

Comment:
It may be better to check `_functionality == FUNC_optimize` in case we set a meaningless number like `2` and we may add other functionalities later.